cookbooks/aws-parallelcluster-environment/recipes/init/restore_home_shared_data.rb

```diff
@@ -18,19 +18,65 @@
   # Restore the shared storage home data if it doesn't already exist
   # This is necessary to preserve any data in these directories that was
   # generated during the build of ParallelCluster AMIs after converting to
-  # shared storage and backed up to a temporary location previously
+  # shared storage and backed up to a temporary location previously.
   # Before removing the backup, ensure the data in the new home is the same
   # as the original to avoid any data loss or inconsistency. This is done
   # by using rsync to copy the data and diff to check for differences.
+  # The diff command excludes files that were originally in /home to focus on
+  # the newly synchronized files. This approach is necessary because the /home
+  # directory may contain pre-existing files such as slurm-*.out generated by
+  # running SLURM jobs and the automatically created lost+found directory.
```

Comment:
why should we exclude these files form the integrity checks?

Reply:
The diff command will check if the two directories are totally the same. So if the target dir(/home) has existing files which are different from backup dir(/tmp/home), the diff command will return exit code 1. That's the reason why integ test `test_shared_home` failed.